eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Utilities/InputServiceMethodExtensions.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using Azure.Core;
+using Microsoft.TypeSpec.Generator.Input;
+using Microsoft.TypeSpec.Generator.Primitives;
+
+namespace Azure.Generator.Management.Utilities
+{
+    internal static class InputServiceMethodExtensions
+    {
+        public static bool IsLongRunningOperation(this InputServiceMethod method)
+        {
+            return method is InputLongRunningServiceMethod || method is InputLongRunningPagingServiceMethod;
+        }
+
+        public static OperationFinalStateVia GetOperationFinalStateVia(this InputServiceMethod method)
+        {
+            if (method is InputLongRunningServiceMethod lroMethod)
+            {
+                return (OperationFinalStateVia)lroMethod.LongRunningServiceMetadata.FinalStateVia;
+            }
+            else if (method is InputLongRunningPagingServiceMethod lroPagingMethod)
+            {
+                return (OperationFinalStateVia)lroPagingMethod.LongRunningServiceMetadata.FinalStateVia;
+            }
+            return OperationFinalStateVia.Location;
+        }
+
+        public static CSharpType? GetResponseBodyType(this InputServiceMethod method)
+        {
+            var operationResponses = method.Operation.Responses;
+            var response = operationResponses.FirstOrDefault(r => !r.IsErrorResponse);
+            var responseBodyType = response?.BodyType;
+            return responseBodyType is null ? null : ManagementClientGenerator.Instance.TypeFactory.CreateCSharpType(responseBodyType);
+        }
+
+        public static CSharpType GetOperationMethodReturnType(this InputServiceMethod method, bool isAsync, CSharpType resourceClientCSharpType, CSharpType resourceDataType)
```

Comment:
this is not in use?

Reply:
It is referenced in two places:
![image](https://github.com/user-attachments/assets/3b252066-0a8b-43a3-a450-6e3a2687a6dd)